### ISSUE TITLE
4.1.1: build fails with pmix 4.1.0

### ISSUE THREAD

**Issue by kloczek** [open-mpi/ompi#9337]
Looks like it is some issue on building ompi 4.1.1 with pmix 4.10.0.
```console
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ompi-4.1.1/opal'
make[2]: Nothing to be done for 'all-am'.
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ompi-4.1.1/opal'
Making all in mca/common/ofi
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ompi-4.1.1/opal/mca/common/ofi'
  CC       common_ofi.lo
In file included from ../../../../opal/mca/pmix/base/base.h:22,
                 from common_ofi.c:29:
common_ofi.c: In function ‘get_package_rank’:
common_ofi.c:321:40: error: ‘PMIX_PACKAGE_RANK’ undeclared (first use in this function)
  321 |     OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_PACKAGE_RANK,
      |                                        ^~~~~~~~~~~~~~~~~
../../../../opal/mca/pmix/pmix.h:153:56: note: in definition of macro ‘OPAL_MODEX_RECV_VALUE_OPTIONAL’
  153 |         if (OPAL_SUCCESS == ((r) = opal_pmix.get((p), (s), &(_ilist), &(_kv)))) {        \
      |                                                        ^
common_ofi.c:321:40: note: each undeclared identifier is reported only once for each function it appears in
  321 |     OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_PACKAGE_RANK,
      |                                        ^~~~~~~~~~~~~~~~~
../../../../opal/mca/pmix/pmix.h:153:56: note: in definition of macro ‘OPAL_MODEX_RECV_VALUE_OPTIONAL’
  153 |         if (OPAL_SUCCESS == ((r) = opal_pmix.get((p), (s), &(_ilist), &(_kv)))) {        \
      |                                                        ^
make[2]: *** [Makefile:1950: common_ofi.lo] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ompi-4.1.1/opal/mca/common/ofi'
```

**Comment by ggouaillardet** [comment]
Duplicate of #9211

**Comment by jsquyres** [comment]
Let's follow up on #9211.
